docs/adding-and-updating-views.rst

```diff
@@ -0,0 +1,189 @@
+5. Run the tests.
+6. If you want your changes to be included in the main Kingfisher Views software, :ref:`push your branch to github and make a pull request<deploy>`.
+
+Example: adding a field
+-----------------------
+
+In this example we're going to add a field to the `tender` and `awards` views. We know from familiarity with the OCDS that `tender` and `award` objects both have :code:`description` fields at their top level, so we would like to see these in the respective views.
+
+1. First, find the query to modify:
+  * We can find the `tender` views in the :code:`005-tender.sql` file.
+  * The main `tender` view is assembled first in the :code:`staged_tender_summary` table.
+
+2. Add the new field to the select query.
+  * You can see the other top level fields in the query, so add it alongside those: 
+
+.. code-block:: sql
+
+    create table staged_tender_summary
+    AS
+    select
+        r.id,
+        r.release_type,
+        r.collection_id,
+        r.ocid,
+        r.release_id,
+        r.data_id,
+        tender ->> 'id' AS tender_id,
+        tender ->> 'title' AS tender_title,
+        tender ->> 'status' AS tender_status,
+        tender ->> 'description' AS tender_description, -- This is our addition
+    ...
+
+3. We can do the same for `awards`, using the :code:`006-awards.sql` file and the :code:`staged_awards_summary_no_data` table:
+
+.. code-block:: sql
+    
+    ...
+    award ->> 'title' AS award_title,
+    award ->> 'status' AS award_status,
+    award ->> 'description' AS award_description, -- This is ours
+    ...
+
+4. Now :ref:`test your changes<testing-changes>`.
+
+5. Then :ref:`add documentation<adding-docs>`. This is **required**!
+
+6. Run the tests to make sure your changes were successful and you didn't break anything else in the process by running: :code:`pytest`
+
+Example: adding an aggregate
+----------------------------
+
+This example demonstrates how Kingfisher Views uses layers of queries and intermediary (:code:`tmp`) tables to build up the final views. In the OCDS data model, various objects can have an array of `documents` attached to them. We're going to add an aggregate (the total number of `documents`) that is missing for one type of object.
+
+1. We're going to be updating the `release summary` views, so we need the :code:`008-release.sql` file.
+  * Queries are in blocks beginning with :code:`drop table if exists` up to :code:`create unique index`.
+  * Queries are grouped together roughly by the stages in the planning process, but there are no hard and fast rules. When you're adding new queries, just try to find a place for them that might make logical sense for the next person who comes along to edit this file - which may well be you.
+
+2. We need to add document counts for `planning` and `tender` objects. They already exist for `award` and `contract`, and the structure is the same, so we can copy one of the existing queries and change the field names, eg:
```

Comment:
> We need to add document counts for `planning` and `tender` objects. 

I think the example only shows adding a document count for the `planning` object?

Reply:
We did both when we planned the docs, but it was getting unwieldy to read. I'll remove mentions of `tender`.